clang-tools-extra/docs/clang-tidy/checks/readability/use-builtin-literals.rst

```diff
@@ -0,0 +1,36 @@
+.. title:: clang-tidy - readability-use-builtin-literals
+
+readability-use-builtin-literals
+================================
+
+Finds literals explicitly casted to a type that could be expressed using builtin prefixes or suffixes.
+
+.. code-block: c++
+
+    (char16_t)U'a'; // -> u'a'
+    (char)'a'; // -> 'a'
+    static_cast<unsigned int>(0x1ul); // -> 0x1u
+    reinterpret_cast<long long int>(3ll); // -> 3LL
+    float(2.); // -> 2.f
+    double{2.}; // -> 2.
```

Comment:
```suggestion
    (char16_t)U'a';                       // -> u'a'
    (char)'a';                            // -> 'a'
    static_cast<unsigned int>(0x1ul);     // -> 0x1u
    reinterpret_cast<long long int>(3ll); // -> 3LL
    float(2.);                            // -> 2.f
    double{2.};                           // -> 2.
```
Maybe it would be visually more pleasing when rendered into HTML if the indications of how stuff is fixed line up appropriately.